python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -1012,6 +1012,23 @@ def resolve_inner(
             "code_location: cannot be used to select assets in user code.",
         )
 
+        # If the code location is in the form of "repo_name@location_name", we need to
+        # split the string and filter the asset keys based on the repository and location name.
+        if "@" in self.selected_code_location:
+            asset_keys = set()
+            location = self.selected_code_location.split("@")[1]
+            name = self.selected_code_location.split("@")[0]
+            for asset_key in cast("RemoteAssetGraph", asset_graph).remote_asset_nodes_by_key:
+                repo_handle = (
+                    asset_graph.get(asset_key)
+                    .resolve_to_singular_repo_scoped_node()
+                    .repository_handle
+                )
+                if repo_handle.location_name == location and repo_handle.repository_name == name:
+                    asset_keys.add(asset_key)
+            return asset_keys
```

Comment:
Seems fine to me because the only other place actually resolving code location asset selections is alerts which gets its strings from the UI.

Reply:
So there is no risk of breaking any existing usages and users can't resolve code location asset selections at definition time since it's not supported

Reply:
Yeah that's my read on it too. The alerts logic relies on serializing this code location asset selection, but no serialized code location will hit this new code path since they can't contain "@"